tests/index/test_index_search_meilisearch.py

```diff
@@ -506,3 +506,11 @@ def test_vector_search(index_with_documents_and_vectors):
         "", opt_params={"vector": [0.1, 0.2], "hybrid": {"semanticRatio": 1.0}}
     )
     assert len(response["hits"]) > 0
+
+
+def test_search_distinct(index_with_documents):
+    index_with_documents().update_filterable_attributes(["genre"])
+    response = index_with_documents().search("How to train your dragin", {"distinct": "genre"})
+    assert isinstance(response, dict)
+    assert len(response["hits"]) == 1
+    assert response["hits"][0]["id"] == "166428"
```

Comment:
Let's modify this test some because searching for `"How to train your dragin"` will always return 1 result even without the distinct attribute. Because of this we won't know if the code actually worked.

At the top of the file we will also need to import `Counter`: `from collections import Counter`

```suggestion
def test_search_distinct(index_with_documents):
    index_with_documents().update_filterable_attributes(["genre"])
    response = index_with_documents().search("with", {"distinct": "genre"})
    genres = dict(Counter([x.get("genre") for x in response["hits"]]))
    assert isinstance(response, dict)
    assert len(response["hits"]) == 11
    assert genres == {None: 9, "action": 1, "Sci Fi": 1}
    assert response["hits"][0]["id"] == "399579"
```

With this modification  we get `{None: 9, 'action': 1, 'Sci Fi': 1}` for the genres, without distinct we would get `{None: 9, 'action': 3, 'Sci Fi': 1}`

Reply:
Thanks for suggesting this test case, this looks better, I should have used a simpler search term in the first place 😅